core/domain/subtopic_page_services.py

```diff
@@ -322,3 +324,100 @@ def delete_subtopic_page(
     subtopic_models.SubtopicPageModel.get(subtopic_page_id).delete(
         committer_id, feconf.COMMIT_MESSAGE_SUBTOPIC_PAGE_DELETED,
         force_deletion=force_deletion)
+
+
+def get_topic_ids_from_subtopic_page_ids(
+    subtopic_page_ids: List[str]
+) -> List[str]:
+    """Returns the topic ids corresponding to the given subtopic page ids.
+
+    Args:
+        subtopic_page_ids: list(str). The ids of the subtopic pages.
+
+    Returns:
+        list(str). The topic ids corresponding to the given subtopic page ids.
+    """
+    topic_ids: List[str] = []
+
+    for subtopic_page_id in subtopic_page_ids:
+        topic_id = subtopic_page_id.split(':')[0]
+        if topic_id not in topic_ids:
+            topic_ids.append(topic_id)
+
+    return topic_ids
+
+
+def get_multi_users_subtopic_pages_progress(
+    user_ids: List[str],
+    subtopic_page_ids: List[str]
+) -> Dict[str, List[subtopic_page_domain.SubtopicPageSummaryDict]]:
+    """Returns the progress of the given user on the given subtopic pages.
+
+    Args:
+        user_ids: list(str). The ids of the users.
+        subtopic_page_ids: list(str). The ids of the subtopic pages.
+
+    Returns:
+        dict(str, list(SubtopicPageSummaryDict)). User IDs as keys and Subtopic
+        Page Summary domain object dictionaries containing details of the
+        subtopic page and users mastery in it as values.
+    """
+
+    topic_ids = get_topic_ids_from_subtopic_page_ids(subtopic_page_ids)
+    topics = topic_fetchers.get_topics_by_ids(topic_ids) # type: ignore[no-untyped-call]
+
+    all_skill_ids_lists = [
+        topic.get_all_skill_ids() for topic in topics if topic
+    ]
+    all_skill_ids = list(
+        {
+            skill_id for skill_list in all_skill_ids_lists
+            for skill_id in skill_list
+        }
+    )
+
+    # Fetch the progress of the all students in all the subtopics assigned
+    # in the group syllabus.
+    all_users_skill_mastery_dicts = (
+        skill_services.get_multi_users_skills_mastery(
+            user_ids, all_skill_ids
+        )
+    )
+
+    all_users_subtopic_prog_summaries: Dict[
+        str, List[subtopic_page_domain.SubtopicPageSummaryDict]
+    ] = {user_id: [] for user_id in user_ids}
+
+    for topic in topics:
+        for subtopic in topic.subtopics:
+            subtopic_page_id = '{}:{}'.format(topic.id, subtopic.id)
+            if subtopic_page_id not in subtopic_page_ids:
+                continue
+            for user_id, skills_mastery_dict in all_users_skill_mastery_dicts.items(): # pylint: disable=line-too-long
```

Comment:
```suggestion
            for user_id, skills_mastery_dict in (
                all_users_skill_mastery_dicts.items()
            ):
```

Try to limit the usage of pylint disables

Reply:
OK, modified it. Thanks!